frontend/wallet/src/render/views/settings/DelegationSettings.re

```diff
@@ -22,9 +22,14 @@ type feeSelection =
   | DefaultAmount
   | Custom(string);
 
+type errorOutcome =
+  | InvalidDelegateKey
+  | Valid;
+
 type modalState = {
   delegate: option(PublicKey.t),
   fee: feeSelection,
+  error: errorOutcome,
```

Comment:
Semantically, it feels better if either this field were called `outcome` or the type were `option(errorOutcome)` so "Valid" would be `None` (ie No error). `error = Valid` is a little in between.